### PR TITLE
kill query action of  store_api

### LINKED ISSUE #1297: [store] support kill action in store
**Summary**

When users execute kill query or query errors occur, we need to call store's kill action in query to destroy query resources.

```
: ) SELECT crashme(sleep(1)) FROM remote_table; -- panic
: ) SELECT *  FROM remote_table WHERE null; -- throw error
: ) KILL QUERY 'query_id';
: ) KILL CONNECTION 'query_id';
```

### PR DESCRIPTION
**Pull request by dantengsky**
I hereby agree to the terms of the CLA available at: https://datafuse.rs/policies/cla/

## Summary

[store_api] blank implementation for kill session action 

For the time being, Store will do nothing for this action; this API is mainly for Query to wire up all the business logics relevant to kill a session.

## Changelog

- New Feature

## Related Issues

Fixes #1297

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

**Comment by dantengsky** [comment]
@zhang2014   please take a look at the kill session API.
it is just an imaginary API, since currently there is nothing for Store to do while killing a session. If anything does not quite fit well, please let me know.

File: common/flights/src/impls/session_api_impl.rs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    pub session_id: String,
+}
+
+#[derive(serde::Serialize, serde::Deserialize, Clone, Debug)]
+pub struct KillSessionReply;
+
+#[async_trait::async_trait]
+impl SessionApi for StoreClient {
+    #[tracing::instrument(level = "debug", skip(self))]
+    async fn kill(&mut self, session_id: String) -> Result<()> {
```
Comment (zhang2014) [inline common/flights/src/impls/session_api_impl.rs:24]: what is session_id? it's query_id in fusequery?

Comment (dantengsky) [reply common/flights/src/impls/session_api_impl.rs:24]: oops, it should be `query_id`, let me rename it

Comment (dantengsky) [reply common/flights/src/impls/session_api_impl.rs:24]: should I rename the method to `kill_query` as well?

Comment (zhang2014) [reply common/flights/src/impls/session_api_impl.rs:24]: Yes, maybe `kill_query` is better.

**Review by zhang2014 — APPROVED** [review]
LGTM 👍

**Pull request merged**